src/saltext/vmware/states/vmc_nat_rules.py

```diff
@@ -0,0 +1,486 @@
+"""
+VMC nat rules state module
+
+Add new nat rule, update existing nat rule and delete existing nat rule from an SDDC.
+
+Example usage :
+
+.. code-block:: yaml
+
+    ensure_nat_rule:
+      vmc_nat_rules.present:
+        - hostname: sample-nsx.vmwarevmc.com
+        - refresh_key: 7jPSGSZpCa8e5Ouks4UY5cZyOtynAhF
+        - authorization_host: console-stg.cloud.vmware.com
+        - org_id: 10e1092f-51d0-473a-80f8-137652c39fd0
+        - sddc_id: b43da080-2626-f64c-88e8-7f31d9d2c306
+        - domain_id: mgw
+        - nat_rule: vCenter_Inbound_Rule_2
+        - verify_ssl: False
+        - cert: /path/to/client/certificate
+        - source_network: "10.117.5.73"
+        - translated_network: "192.168.1.1"
+
+
+.. warning::
+
+    It is recommended to pass the VMC authentication details using Pillars rather than specifying as plain text in SLS
+    files.
+
+
+"""
+import logging
+
+from saltext.vmware.utils import vmc_constants
+from saltext.vmware.utils import vmc_state
+
+log = logging.getLogger(__name__)
+NAT_RULE_NOT_FOUND_ERROR = "could not be found"
+
+
+def present(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    tier1,
+    nat,
+    nat_rule,
```

Comment:
**suggestion**: follow @dhiltonp's advice - if the `nat_rule` is the primary identifier, the `name` parameter is what should be used+documented instead. See https://github.com/saltstack/salt-ext-modules-vmware/pull/117/files#r727542751 for more info.

Reply:
As discussed, changes will be made as part of the issue: https://github.com/saltstack/salt-ext-modules-vmware/issues/120 for all states